ctc_lib/container_tool_command_base.go

```diff
@@ -67,3 +76,32 @@ func (ctb *ContainerToolCommandBase) ReadTemplateFromFlagOrCmdDefault() string {
 	}
 	return flags.TemplateString
 }
+
+// The command line flags are not parsed until Execute is called.
+// Hence all logic to init global flags should happen in PersistentPreRun.
+func (ctb *ContainerToolCommandBase) initGlobalFlags() {
+	init_persisted_flags := func(cmd *cobra.Command, args []string) {
+		Log.SetLevel(flags.LogLevel.Level)
+	}
+	// Set PersistentPreRun to above function if its not set already.
+	// Else overwrite the existing one with init_persisted_flags followed by
+	// user set.
+	if ctb.PersistentPreRun == nil && ctb.PersistentPreRunE == nil {
```

Comment:
It might be cool to support existing PreRuns as well. You could do something like:

```go
originalPrerun := ctb.PersistentPrerun

ctb.PersistentPrerun = func(cmd *cobra.Command, args []string) {
  originalPrerun()
  initPersistentFlags()
}
```

Reply:
Actually i just found out, i dont need this any more. 
I registered a function using cobra.OnInitialize() which is called after Command.ParseFlags().
its changed in https://github.com/GoogleCloudPlatform/runtimes-common/pull/564. Will make that change here too and update the PR.